sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -242,21 +241,21 @@ public override void Dispose()
 
             readonly static HttpMethod s_patch = new HttpMethod("PATCH");
 
-            public static HttpMethod ToHttpClientMethod(HttpPipelineMethod method)
+            public static HttpMethod ToHttpClientMethod(RequestMethod method)
             {
-                switch (method)
+                switch (method.Method)
                 {
-                    case HttpPipelineMethod.Get:
+                    case "GET":
                         return HttpMethod.Get;
-                    case HttpPipelineMethod.Post:
+                    case "POST":
                         return HttpMethod.Post;
-                    case HttpPipelineMethod.Put:
+                    case "PUT":
                         return HttpMethod.Put;
-                    case HttpPipelineMethod.Delete:
+                    case "DELETE":
                         return HttpMethod.Delete;
-                    case HttpPipelineMethod.Patch:
+                    case "PATCH":
                         return s_patch;
-                    case HttpPipelineMethod.Head:
+                    case "HEAD":
                         return HttpMethod.Head;
 
                     default:
```

Comment:
Should we support everything HttpMethod does since you can call the RequestMethod ctor to create other values?